### PR TITLE
Improve error message for unavailable plugins.

### LINKED ISSUE #16: Error: Not available. codefmt doesn't have a default formatter for this buffer.
[The announcement](http://google-opensource.blogspot.ru/2015/01/format-code-easy-way-with-codefmt.html) claims this plugin supports cpp & js natively.
so after installing with vundle ( as provided in the readme file )
when I try to `:FormatCode` on .cpp or .js files I get following error:
`Not available. codefmt doesn't have a default formatter for this buffer.`

What I am missing? Should one use VAM or install any additional plugins? Thanks in advance.

### PR DESCRIPTION
**Pull request by fowles**
Fixes #16

### CODE REVIEW

File: autoload/codefmt.vim [hunk 1]
```diff
@@ -279,9 +282,23 @@ function! s:GetFormatter(...) abort
...
       let l:formatter = l:default_formatters[0]
     else
-      call maktaba#error#Shout(
-          \ 'Not available. codefmt doesn''t have a default formatter for ' .
-          \ 'this buffer.')
+      " Check if we have formatters that are not avialable for some reason.
+      " Report a better error message in that case.
+      let l:unavailable_formatters = filter(
+            \ copy(l:formatters), 'v:val.AppliesToBuffer()')
+      let l:error = ''
```
Comment (dbarnett) [inline autoload/codefmt.vim]: You can replace this whole thing with `let l:error = join(map(copy(l:unavailable_formatters), 's:GetSetupInstructions(v:val)'), "\n")`. Doesn't have to be all one line, but using `join()` would be a big improvement, and `map()` probably would too.

File: vroom/main.vroom [hunk 2]
```diff
@@ -18,7 +18,30 @@ stub that out for vroom.
   :endfunction
   :call maktaba#test#Override('repeat#set', 'FakeRepeat')
 
-We'll also stub it out to not care whether certain executables are installed on
+
+Before we stub out availability checks, test :FormatCode for unavailable plugins
```
Comment (dbarnett) [inline vroom/main.vroom]: Hmm, putting this gnarliness at the very beginning of the file would undermine the sequence of concepts I'm trying to build in #11. Could we just add a `codefmt#ReenableIsAvailableChecksForTesting()` or `codefmt#OverrideIsAvailableChecksForTesting({is_enabled})` and put further down in the file where it's more natural?

Comment (fowles) [inline vroom/main.vroom]: Actually, I have spent the last hour or so trying to make that work.  I clearly don't understand something about vim's evaluation of globals or something.  Do you mind taking a crack at it?

Comment (dbarnett) [inline vroom/main.vroom]: Heh, I know how you feel. Sure, if you want to just address the other comment and submit, I can take care of it.

**Comment by dbarnett** [comment]
LGTM modulo the loop cleanup.

**Pull request merged**